Package.swift

```diff
@@ -116,6 +116,18 @@ func addIntegrationTestTarget(_ name: String) {
     ]
 }
 
+func addCustomIntegrationTestTarget(_ folder: String, _ name: String, dependsOn dependency: String) {
```

Comment:
Not sure I want to add this complexity to the package manifest.  Maybe better to locate these tests with S3 since that is the service being authenticated in the test?

In any case, this file is overwritten and can't be hand-modified.  See the `AWSSDKSwiftCLI` subproject for where this file is generated:
https://github.com/awslabs/aws-sdk-swift/tree/main/AWSSDKSwiftCLI

Reply:
Yup I was just going through that. The issue with not modifying the IntegrationTestTarget function is that right now its setup so that the name of the folder corresponds to the test dependency. So AWSS3IntegrationTests depends on AWSS3. However ProcessCredentialsIntegrationTests depends on AWSS3 so that setup will not work even if its put under Services.

Reply:
I was going to revert my changes to Package.swift and discuss options with you. One workaround is we could put this test file inside of AWSS3

Reply:
Since different credential providers may be tested against different services depending on use-case perhaps it does make sense to do that. I have changed it in my latest commit.